### PR TITLE
Onward journey metadata display updates

### LINKED ISSUE #9946: Optimo Editorial OJs metadata updates
**Is your feature request related to a problem? Please describe.**
REF: NEWSWORLDSERVICE-1366

To give users more depth and analysis about related stories and content, we want to extend the existing Editorial Onward Journey (horizontal scrollable) to include more metadata about each of the linked content.

AreS has been updated to return the following enriched data for these OJs:

- Date (timestamp)

Example of the data can be found [here](https://www.bbc.com/kyrgyz/articles/c88ld5g4xxxo.json), found under the types `aresLink` and `aresMedia`.

Editors would like to display the 'overtyped' title that will sit above the OJs. This is an optional field, displaying nothing if no title is supplied.

Editors would also like the number of lines increased to 4 for the OJ blocks.

**Describe the solution you'd like**

Update the Optimo Editorial OJs (horizontal scrollable promo) to include the new data returned from AreS:
- Add the Timestamp to the block (if available)
- Increase the number of lines displayed up to 4 lines
- Add the overtyped title above the OJ blocks (if available)

Update the related content promo (when the links block is last block in article) 
- Update the related content block to support timestamp if available
- Overtyped title is already supported

UX: https://zpl.io/O0RgxxQ

Further notes and full a11y Acceptance Criteria can be found in https://jira.dev.bbc.co.uk/browse/NEWSWORLDSERVICE-1366

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by amoore108**
Resolves https://github.com/bbc/simorgh/issues/9946

**Overall change:**
Adds more metadata (timestamp and overtyped title) to the scrollable promo and related content components on Article pages. Scrollable promo has also increased the line cut off from 3 lines to 4 lines.

**Code changes:**
- Adds the timestamp to scrollable and related content promos, extracted from the `aresLink` field in the AreS response
- Adds the overtyped title to the scrollable promo if its available
- Updates the scrollable promo to use the Psammead `Link` component, so that it has the same hover and click behaviour as the other promos

**2 Dev a11y swarm required**
- Test new landmark for Scrollable promo
- Test title is read correctly for Scrollable promo
- Test timestamp is read correctly for Scrollable promo

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/ArticleRelatedContent/index.jsx [hunk 1]
```diff
@@ -55,12 +56,21 @@ export const buildStoryPromos = optimoRelatedContent => {
...
+      const aresLinkBlock = filterForBlockType(
+        pathOr({}, ['model', 'blocks'], item),
+        'aresLink',
+      );
+      const timestamp = path(
+        ['model', 'blocks', '0', 'model', 'timestamp'],
+        aresLinkBlock,
+      );
+
+      const headingTag = timestamp ? 'h3' : 'div';
```
Comment (greenc05) [inline src/app/containers/ArticleRelatedContent/index.jsx:68]: Is this for the onwards journeys?

File: src/app/components/ScrollablePromo/index.jsx [hunk 2]
```diff
@@ -41,12 +69,42 @@ const ScrollablePromo = ({ blocks, blockGroupIndex }) => {
...
+    role: 'region',
+    'aria-label': ariaLabel,
+  };
+
   return (
-    <GridItemMediumNoMargin>
+    <GridItemMediumNoMargin {...a11yAttributes}>
+      {titleBlock && (
+        <LabelComponent
+          id={titleId}
```
Comment (DarioR01) [inline src/app/components/ScrollablePromo/index.jsx]: Should this match the `aria-label` used in `GridItemMediumNoMargin`?

**Review by DarioR01 — APPROVED** [review]

File: src/app/components/ScrollablePromo/index.jsx (hunk 2 above)
Comment (amoore108) [reply src/app/components/ScrollablePromo/index.jsx]: Good question. Should it be `aria-labelledby` if the `titleBlock` value is available (meaning the `<LabelComponent />` will be there) and `aria-label` if `titleBlock` isn't available? cc @greenc05

File: src/app/components/ScrollablePromo/index.jsx [hunk 3]
```diff
@@ -41,12 +69,42 @@ const ScrollablePromo = ({ blocks, blockGroupIndex }) => {
...
+    path(
+      ['0', 'model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'text'],
+      blocks,
+    );
+
   const blocksWithoutTitle = blocks[0].type === 'title' ? tail(blocks) : blocks;
 
   const isSingleItem = blocksWithoutTitle.length === 1;
 
+  const titleId = 'eoj-recommendations-heading';
```
Comment (greenc05) [inline src/app/components/ScrollablePromo/index.jsx:83]: should this be a more generic name? 'eoj-recommendations-heading'

File: src/app/components/ScrollablePromo/index.jsx (hunk 2 above)
Comment (amoore108) [reply src/app/components/ScrollablePromo/index.jsx]: This has now been updated using the above logic, thanks @greenc05 🙏🏼

**Review by mfonofm — APPROVED** [review]
LGTM  :+1:

**Pull request merged**